README.md

```diff
@@ -138,9 +137,121 @@ The best way to test this service with simulated data is to use **PostMan** to s
     500.568
     ```
 
+## REST Client
+
+This device service allows EdgeX to send GET/PUT request to end device.
+End device will respond to GET and PUT requests received from EdgeX. For this end device will be running REST server.
+
+### Configuration
+
+To use this device service, device profile and device file needs to be configured. 
+This device service use the standard configuration defined by the **Device SDK**. 
+
+The `DeviceList` configuration is standard except that it is mandatory to provide end device parameters in the `DeviceList.Protocols.EndDevice_Params` structure. The following is a sample `DeviceList` that works with the sample device profiles referenced below. `path` parameter is optional.
+
+```toml
+[[DeviceList]]
+  Name = "2way-rest-device"
+  ProfileName = "sample-2way-rest-device"
+  Description = "RESTful Device that sends data"
+  Labels = [ "rest", "2way-rest-device" ]
+  [DeviceList.Protocols]
+    [DeviceList.Protocols.REST]
+	  Host = "127.0.0.1"
+	  Port = "5000"
+	  Path = "api"
+  [[DeviceList.AutoEvents]]
+    Interval = "20s"
+    OnChange = false
+    SourceName = "json"
+```
+
+### Device Profile
+
+This section shows sample device profile defined for REST client functionality.
+
+- [**sample-2way-rest-device**](cmd/res/profiles/sample-2way-rest-device.yaml)
+
+### Testing/Simulation of REST client
+
+This device service supports commanding functionality for the data types as shown in below table.
+
+| Data Type | GET                | PUT                |
+|---------	|------------------- |------------------- |
+| Binary	| :heavy_check_mark: | :X				  |
+| Object	| :heavy_check_mark: | :heavy_check_mark: |
+| Bool	   	| :heavy_check_mark: | :heavy_check_mark: |
+| String	| :heavy_check_mark: | :heavy_check_mark: |
+| Uint8		| :heavy_check_mark: | :heavy_check_mark: |
+| Uint16	| :heavy_check_mark: | :heavy_check_mark: |
+| Uint32	| :heavy_check_mark: | :heavy_check_mark: |
+| Uint64	| :heavy_check_mark: | :heavy_check_mark: |
+| Int8	    | :heavy_check_mark: | :heavy_check_mark: |
+| Int16	    | :heavy_check_mark: | :heavy_check_mark: |
+| Int32	    | :heavy_check_mark: | :heavy_check_mark: |
+| Int64	    | :heavy_check_mark: | :heavy_check_mark: |
+| Float32   | :heavy_check_mark: | :heavy_check_mark: |
+| Float64   | :heavy_check_mark: | :heavy_check_mark: |
+
+Using `curl` command-line utility or `PostMan` we can test commanding functionaity of the REST device service.
```

Comment:
Please explain how to use these tools to provide the 2way rest device that is responding to the GET and PUT requests

Reply:
@lenny-intel , We are using nodejs based REST emulator as end devices for testing this. CURL commands for sending GET/PUT request to EdgeX.

Reply:
> We are using nodejs based REST emulator as end devices for testing this

Great, can you include this in a subfolder with directions on using it?